site/src/api/errors.ts

```diff
@@ -61,12 +61,18 @@ export const mapApiErrorToFieldErrors = (
 export const getErrorMessage = (
   error: unknown,
   defaultMessage: string,
-): string =>
-  isApiError(error)
-    ? error.response.data.message
-    : error instanceof Error
-    ? error.message
-    : defaultMessage;
+): string => {
+  if (isApiError(error)) {
+    return error.response.data.message;
+  }
+  if (error instanceof Error) {
+    return error.message;
+  }
+  if (typeof error === "string") {
+    return error;
+  }
+  return defaultMessage;
```

Comment:
I changed it because we depend on it being a string for job build failures, and we've just been silently mishandling those for a long time. I can remove the `instanceof Error` bit tho.